asv_bench/benchmarks/benchmarks.py

```diff
@@ -39,8 +39,8 @@
 
 class BaseTimeGroupBy:
     def setup(self, shape, ngroups=5, groupby_ncols=1):
-        if callable(ngroups):
-            ngroups = ngroups(shape[0])
+        if isinstance(ngroups, str) and ngroups == "huge_amount_groups":
+            ngroups = min(shape[0] // 2, 5000)
```

Comment:
```suggestion
        if not isinstance(ngroups, int):
            ngroups = self.ngroups_size[ngroups](shape[0])
```
For better maintainability, I would add a class dict with lambdas like it's done [here](https://github.com/modin-project/modin/blob/d1321f5fc7a0c5dd755f6f3c260ea53d54206119/asv_bench/benchmarks/benchmarks.py#L396-L400) with subset param.

We also should change [`BaseTimeValueCounts`](https://github.com/modin-project/modin/blob/d1321f5fc7a0c5dd755f6f3c260ea53d54206119/asv_bench/benchmarks/benchmarks.py#L396) setup logic, since it uses `ngroups` parameter from `GROUPBY_NGROUPS` as well. I think we should inherit `BaseTimeValueCounts` from `BaseTimeGroupBy` and replace frame generation in setup function with:
```python
super().setup(shape=shape, ngroups=ngroups, groupby_ncols=ncols)
```